plugin/evm/block_manager.go

```diff
@@ -24,28 +23,17 @@ var (
 	apricotPhase1MinGasPrice = big.NewInt(params.ApricotPhase1MinGasPrice)
 )
 
-type blockManager struct {
-	blockExtension extension.BlockManagerExtension
-	vm             *VM
-}
-
-func newBlockManager(vm *VM, blockExtension extension.BlockManagerExtension) *blockManager {
-	return &blockManager{
-		blockExtension: blockExtension,
-		vm:             vm,
-	}
-}
-
 // newBlock returns a new Block wrapping the ethBlock type and implementing the snowman.Block interface
-func (bm *blockManager) newBlock(ethBlock *types.Block) (*Block, error) {
+func (vm *VM) newBlock(ethBlock *types.Block) (*Block, error) {
```

Comment:
nit: could we have this as `func newBlock(ethBlock *types.Block, vm *VM) (*Block, error)`? Not too sure why it would be a method on `VM`.

Also second nit, I would rename `ethBlock` to `block` or `b` given it doesn't conflict with a `block` I think, and is not really a eth-block (aka libevm block)

Reply:
I agree that we can pass vm as a param (in case we want to rather pass it as a interface).

Eth block is the eth block (it's core/types.Block), not sure what you meant there.

Reply:
hum. I'm still not done here. appreciate the feedback though and will consider it.